packages/shared/src/contexts/payment/StoreKit.tsx

```diff
@@ -118,6 +150,58 @@ export const StoreKitSubProvider = ({
     [user?.subscriptionFlags?.appAccountToken],
   );
 
+  useEffect(() => {
+    const eventName = 'iap-purchase-event';
+    promisifyEventListener<void, PurchaseEvent>(
+      eventName,
+      (event) => {
+        const { name, detail, product } = event.detail;
+        switch (name) {
+          case PurchaseEventName.PurchaseCompleted:
+            logRef.current({
+              event_name: LogEvent.CompleteCheckout,
+              extra: {
+                user_id: user?.id,
+                cycle: productIds[product.attributes.offerName].duration,
+                localCost: product.attributes.offers[0].price,
+                localCurrency: product.attributes.offers[0].currencyCode,
+                payment: SubscriptionProvider.AppleStoreKit,
+              },
+            });
+            router.push(plusSuccessUrl);
+            break;
+          case PurchaseEventName.PurchaseInitiated:
+            logRef.current({
+              event_name: LogEvent.InitiatePayment,
+            });
+            break;
+          case PurchaseEventName.PurchaseFailed:
+            logRef.current({
+              event_name: LogEvent.ErrorCheckout,
+              extra: {
+                errorCode: detail,
+              },
+            });
+            displayToast(DEFAULT_ERROR);
+            break;
+          case PurchaseEventName.PurchasePending:
```

Comment:
@idoshamun do you want an event for when the purchase is pending, i.e. user has "Ask to buy" activated

Reply:
what's the difference with initiate checkout?

Reply:
initiate checkout occurs when the user click on the continue button to initiate the purchase.
Purchase pending occurs when the user wants to buy it, but requires approval first. [Ask to buy](https://support.apple.com/en-us/105055)

Reply:
And it's not really purchase complete, as it can be approved or reject. If the user is still on the same page while it gets approved, it will send the purchase complete event and redirect them to the success page. Otherwise it'll happen in the background, but the server event will be fired when it gets approved, and user gets their plus.